apm-agent-core/src/main/java/co/elastic/apm/agent/bci/ElasticApmAgent.java

```diff
@@ -845,4 +873,12 @@ public static ClassLoader getInstrumentationClassLoader(String adviceClass) {
         }
         return classLoader;
     }
+
+    public static Collection<String> getPluginClassLoaderRootPackages(String pluginPackage) {
+        Collection<String> pluginPackages = pluginPackages2pluginClassLoaderRootPackages.get(pluginPackage);
+        if (pluginPackages != null) {
+            return pluginPackages;
+        }
+        return Collections.singleton(pluginPackage);
+    }
```

Comment:
This provides either the configured root packages if there are such (typically the aren't) OR the plugin base package, but not a superset of both. How come it works?

Reply:
The configured root packages need to contain the base package. That's because plugins can choose to not include the base package to the root packages. That's done in the concurrent plugin which is all loaded by the agent CL.

Reply:
OK, that's what I missed. 
I think it would be much more intuitive to merge the two instead of introducing this requirement.
The java concurrent plugin can simply return `null`, which will be taken into account when merging.

Reply:
You mean that `null` would serve as a special value that removes the plugin package? I think it's clearer to not allow null values and just use the value returned by the customizer. If the customizer is not used, it just uses the plugin package by default.
I've clarified the docs of the customizer that the plugin package has to be explicitly included.